// perfume.js/dist/lib/reportPerf.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var config_1 = require("./config");
var getNavigatorInfo_1 = require("./getNavigatorInfo");
var onVisibilityChange_1 = require("./onVisibilityChange");
var utils_1 = require("./utils");
var vitalsScore_1 = require("./vitalsScore");
/**
 * Sends the User timing measure to analyticsTracker
 */
exports.reportPerf = function (measureName, data, customProperties) {
    utils_1.pushTask(function () {
        // Doesn't send timing when page is hidden
        if ((onVisibilityChange_1.visibility.isHidden && measureName.indexOf('Final') < 0) ||
            !config_1.config.analyticsTracker) {
            return;
        }
        // Send metric to custom Analytics service
        config_1.config.analyticsTracker({
            metricName: measureName,
            data: data,
            eventProperties: customProperties || {},
            navigatorInformation: getNavigatorInfo_1.getNavigatorInfo(),
            vitalsScore: vitalsScore_1.getVitalsScore(measureName, data),
        });
    });
};
